docs/source/metric_integrations.mdx

```diff
@@ -0,0 +1,136 @@
+# Using 🤗 Evaluate in other frameworks
+
+This guide outlines how `evaluate` metrics can be used with popular machine learning frameworks.
+
+## 🤗 Transformers
+
+To run the 🤗 Transformers examples make sure you have installed the following libraries:
+
+```bash
+pip install datasets transformers torch evaluate nltk rouge_score
+```
+
+### Trainer
+
+The metrics in `evaluate` can be easily integrated in the [`~transformers.Trainer`]. The `Trainer` accepts a `compute_metrics` keyword argument where a function to compute metrics can be passed. One can specify the evaluation interval with `evaluation_strategy` in the [`~transformers.TrainerArguments`] and based on that the model is applied to the evaluation said and the predictions and labels passed to `compute_metrics`.
+
+```python
+from datasets import load_dataset
+from transformers import AutoTokenizer, AutoModelForSequenceClassification, TrainingArguments, Trainer
+import numpy as np
+import evaluate
+
+# Prepare and tokenize dataset
+dataset = load_dataset("yelp_review_full")
+tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")
+
+def tokenize_function(examples):
+    return tokenizer(examples["text"], padding="max_length", truncation=True)
+
+tokenized_datasets = dataset.map(tokenize_function, batched=True)
+
+small_train_dataset = tokenized_datasets["train"].shuffle(seed=42).select(range(1000))
+small_eval_dataset = tokenized_datasets["test"].shuffle(seed=42).select(range(1000))
+
+# Setup evaluation with 
+metric = evaluate.load("accuracy")
+
+def compute_metrics(eval_pred):
+    logits, labels = eval_pred
+    predictions = np.argmax(logits, axis=-1)
+    return metric.compute(predictions=predictions, references=labels)
+
+# Load pretrained model and evaluate model after each epoch
+model = AutoModelForSequenceClassification.from_pretrained("bert-base-cased", num_labels=5)
+training_args = TrainingArguments(output_dir="test_trainer", evaluation_strategy="epoch")
+
+trainer = Trainer(
+    model=model,
+    args=training_args,
+    train_dataset=small_train_dataset,
+    eval_dataset=small_eval_dataset,
+    compute_metrics=compute_metrics,
+)
+
+trainer.train()
+```
+
+### Seq2SeqTrainer
+
+For sequence-to-sequence tasks such as translation or summarization the [`~transformers.Seq2SeqTrainer`] is useful. It allows for the use of the generate method when setting `predict_with_generate=True`. That means we evaluate generated text within the `compute_metric` function, we just need to decode the predictions and labels first.
```

Comment:
It was a bit hard to understand this paragraph with the eyes of a new reader. Why is the `predict_with_generate` needed? What would be used if not?

Reply:
I tried to elaborate a bit more.